examples/plot_0_sin.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+"""
+Introductory example - Plotting sin
+===================================
+
+This is a general example demonstrating a Matplotlib plot output, embedded
+rST, the use of math notation and cross-linking to other examples. It would be
+useful to compare the :download:`source Python file <plot_0_sin.py>` with the
+output below.
+
+Example files should start with a header docstring. Anything before the
+docstring is ignored by Sphinx-Gallery and will not appear in the rendered
+output, nor will it be executed. This docstring requires an rST header, used
+at the title of the example and to correctly build reference links.
+
+Once you close the docstring you will be writing Python code. This code gets
+executed by Sphinx-Gallery and any output, including plots will be captured.
+You can also alternate between code blocks and text blocks that provide
+explanations about the code.
+
+To include displayed math notation, use the directive ``.. math::`` and to
+include inline math notation use ``:math:``. For example, we wish to plot the
+following function:
+
+.. math::
+
+    x \\rightarrow \\sin(x)
+
+Here the function :math:`\\sin` is evaluated at each point the variable
+:math:`x` is defined. When writting latex in a Python string, ensure that you
+escape the backslashes or use a raw docstring. You do not need to do
+this in text blocks (see below).
+"""
+
+# Code source: Óscar Nájera
+# License: BSD 3 clause
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+x = np.linspace(0, 2 * np.pi, 100)
+y = np.sin(x)
+
+plt.plot(x, y)
+plt.xlabel('$x$')
+plt.ylabel('$\sin(x)$')
+# To avoid matplotlib text output
+plt.show()
```

Comment:
This shouldn't be necessary; you could delete these two lines if the prior line was an assignment (`_ = plt.ylabel('$\sin(x)$')`).

Reply:
True, but it would be good to still make a comment on why I am assigning it (so it would only be deleting 1 line) - and then I should probably be consistent and change the other examples. Any reason you are opposed to `plt.show()`?

Reply:
ping @drammock 
I will amend `plot_capture_repr` to show the text output and the two ways of avoiding it. I do think I should otherwise be consistent in all the examples though.

Reply:
Good point about needing to explain it either way. And there is value in being consistent.  I guess my feeling is that if we consistently do `plt.show()` then people will assume it's required (which it's not). But since it's currently consistent, and you added a good explanation in the `capture_repr` file, let's call it good enough.  If it still bothers me in a week I'll do a new PR to address it.